### PR TITLE
performance: use lazy loading for components

### PR DESCRIPTION
**Pull request by nhsz**
This PR applies [lazy-loading](https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading) for the following components, to reduce initial JS payload and reduce loading time

- `Codeblock`
- `CodeModal`
- `FeedbackCard`
- `FeedbackWidget`
- `GlossaryDefinition`
- `GlossaryTooltip`
- `Modal`
- `SimulatorModal`
- `Tooltip`

### CODE REVIEW

**Review by wackerow — APPROVED** [review]
Lgtm! Loading as expected, maybe just a slight delay on the first load if you go straight to it, but I'm not worried about that given it's location on the page. Nice job! 

Wondering if we should just make a helper function for this to lazy load with `{ssr: false}`

**Review by pettinarip — COMMENTED** [review]
As discussed offline, I still think there is no good reason to lazily load content components from this list at this early stage of the epic.

By content components I'm referring to `CommunityEvents` and `CalloutBanner`. The contents of these components are visible on the first render (they are included in the html file). If we lazy load them, the content will not be visible to crawlers, which is one of the main reasons we do static rendering of this page.

Also, these components do not have heavy deps. On the contrary, they seem to be pretty small units of code. So I wouldn't expect much difference in perf by including them.

The `CodeModal` & `Codeblock` are good examples of two components that are great candidates for lazy loading, since they are not visible on the first render and may have other external deps.

File: src/components/Simulator/SimulatorModal.tsx [hunk 1]
```diff
@@ -16,13 +16,15 @@ type SimulatorModalProps = ModalContentProps &
...
 
-export const SimulatorModal = ({
+const SimulatorModal = ({
   isOpen,
   onClose,
   children,
   size,
   ...restProps
 }: SimulatorModalProps) => {
+  console.log("LOADING MODAL")
```
Comment (pettinarip) [inline src/components/Simulator/SimulatorModal.tsx:26]: ✂️

**Review by pettinarip — COMMENTED** [review]
@nhsz from the code side, it looks good, but don't really know how this is impacting the overall performance. Have you done any analysis? or do you have a bundle size comparison?

File: src/components/Simulator/SimulatorModal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nhsz) [reply src/components/Simulator/SimulatorModal.tsx:26]: we don't, debuggin dust

**Review by corwintines — APPROVED** [review]

**Review by pettinarip — CHANGES_REQUESTED** [review]
@nhsz I've ran some tests on this with the [bundle analyzer](https://nextjs.org/docs/app/building-your-application/optimizing/bundle-analyzer) and compare them with `dev`.

I've added the results here:
https://docs.google.com/spreadsheets/d/1AKS_h0_xGxyI6L4X9TexJ42IFLkxw6Kc3vgkw5Celfc/edit?usp=sharing

The main `_app` file is reduced by ~5kb, which is less than 1% improvement for this bundle (the main source of problems). In addition, these changes create 10 more chunks, which is neither good nor bad, but they add more network requests and redundant payload to each chunk, making the total bundle size even larger than the previous one.

I'm not against using lazy load, but I would use it with caution since we might be adding complexity for no good reason.

For example, there are a few components (Tooltip, GlossaryDefinition) in this list that are less than 4kb uncompressed. So creating a new chunk for them might not seem worthwhile.

IMO there is no significant difference in the bundle sizes. The optimization in general doesn't justify adding more complexity to the codebase.

Not sure the next steps here but lmk your thoughts.

**Comment by minimalsm** [comment]
Closing this out per @pettinarip's comment and the internal discussion around this change. Please re-open if you disagree with my understanding here @nhsz

**Pull request closed without merging**